### DISCUSSION TITLE
Geomap Panel Feedback

### DISCUSSION THREAD

**Discussion by drew08t** [grafana/grafana#62159] [Dashboards, Panels & Visualizations]
Hello everyone! As our team at Grafana continues development of the [Geomap Panel](https://grafana.com/docs/grafana/latest/panels-visualizations/visualizations/geomap/), we look to you, the community, for feedback and guidance. For those not familiar with it, the Geomap panel allows you to view and customize the world map using geospatial data. You can configure various overlay styles and map view settings to easily focus on the important location-based characteristics of the data. Read more in the [documentation](https://grafana.com/docs/grafana/latest/panels-visualizations/visualizations/geomap/).

![image](https://user-images.githubusercontent.com/60050885/214693168-ea20c875-6e6c-4de2-93dd-ba5f329318d0.png)
![image](https://user-images.githubusercontent.com/60050885/214692737-721e88dc-c955-4793-92e1-68a4b20ddfe5.png)
![image](https://user-images.githubusercontent.com/60050885/214693271-9d80ac84-b8aa-41be-8703-0d1c799eb01b.png)
![image](https://user-images.githubusercontent.com/60050885/214692757-a00def94-9880-449b-9d80-6b54a500938d.png)

## Why Discussion?
We’ve planned additional features and design elements for future releases to make Geomap panels even more powerful tools for visualizing geospatial data. We are adding this discussion to see what use cases are important to the community. We are looking forward to hearing from you!

**Comment by chim678** [comment]
Hello!

Thank you for the plugin. Looks very promising and useful for our projects. We need to collect geospatial data about different equipment, often from different data sources, which brings the need for some additional features. 

1) Would be nice to have the next feature - a check button (in the legend for example) that hides or unhides layers. Maybe there is already a feature like that or similar, but I have not found anything in the documentation. I consider that feature very useful since sometimes panels may be overwhelmed with markers and a UI, that hides some layers, helps a lot.

2) And the second, maybe, it's more like a small bug. We try to use consistent data fetched from the queries (unambiguously defined columns for latitude and longitude, correct data format and etc.). But still, sometimes we need to transform data and after the transformation, we see an undefined RefID in the Data field of the editor, it still works though.

Thank you.

**Reply by drew08t** [reply]
> Hello!
> 
> Thank you for the plugin. Looks very promising and useful for our projects. We need to collect geospatial data about different equipment, often from different data sources, which brings the need for some additional features.
> 
> 1. Would be nice to have the next feature - a check button (in the legend for example) that hides or unhides layers. Maybe there is already a feature like that or similar, but I have not found anything in the documentation. I consider that feature very useful since sometimes panels may be overwhelmed with markers and a UI, that hides some layers, helps a lot.
> 2. And the second, maybe, it's more like a small bug. We try to use consistent data fetched from the queries (unambiguously defined columns for latitude and longitude, correct data format and etc.). But still, sometimes we need to transform data and after the transformation, we see an undefined RefID in the Data field of the editor, it still works though.
> 
> Thank you.

I'm happy to hear you are finding the panel useful and definitely appreciate the feedback, please keep it coming!

1. This is a great suggestion. I believe we have something similar on our roadmap. I'll make sure your suggestion gets included!
2. I will attempt to replicate this behavior on my end. Do you have a representative dataset you can provide for testing?

Thanks!

**Reply by chim678** [reply]
Hello!

Thank you very much for the reply. 
Yes, I can provide a dataset. This is the link to the dashboard in GrafanaCloud:
https://chimrovav.grafana.net/d/xenaLFxVz/geomap-test?orgId=1
I could also provide the needed permission if it's necessary. 

But overall, I created two queries with CSV content of the TestData DB. Defined next columns and rows: Query A - name, latitude
a,10; Query B - name, latitude b,10. Then in the Transform, I concatenated them with 'Copy frame name to field name'. And then, in the data field, you can see the concatenated query with the undefined RefID.

Thanks.

**Reply by chim678** [reply]
> > Hello!
> > Thank you for the plugin. Looks very promising and useful for our projects. We need to collect geospatial data about different equipment, often from different data sources, which brings the need for some additional features.
> > 
> > 1. Would be nice to have the next feature - a check button (in the legend for example) that hides or unhides layers. Maybe there is already a feature like that or similar, but I have not found anything in the documentation. I consider that feature very useful since sometimes panels may be overwhelmed with markers and a UI, that hides some layers, helps a lot.
> > 2. And the second, maybe, it's more like a small bug. We try to use consistent data fetched from the queries (unambiguously defined columns for latitude and longitude, correct data format and etc.). But still, sometimes we need to transform data and after the transformation, we see an undefined RefID in the Data field of the editor, it still works though.
> > 
> > Thank you.
> 
> I'm happy to hear you are finding the panel useful and definitely appreciate the feedback, please keep it coming!
> 
> 1. This is a great suggestion. I believe we have something similar on our roadmap. I'll make sure your suggestion gets included!
> 2. I will attempt to replicate this behavior on my end. Do you have a representative dataset you can provide for testing?
> 
> Thanks!

Hello! Any updates regarding the features in the discussion?

**Comment by andnofence** [comment]
I just upgraded Grafana to 9.4.2 and have explored the new Geomap plugin very quickly.
1. I strongly second @chim678's suggestion about having layers that can be turned on and off. It is very useful to visualize different layers in the same map and having the option to turn them on and off.
2. Dynamic GeoJSON wasn't what I expected. I have polygons stored in the database that I want to render on the map and I was expecting to be able to do this, instead it just seems like a way to dynamically adjust how the pre-defined GeoJSON files end up looking. Getting GeoJSON polygons from my database and into the map would allow me to get rid of a custom built plugin that can do this (but it's otherwise not very good). Is this feature on your roadmap?

Speaking of roadmaps, is the GeoMap roadmap publicly available?

I'm very happy to see that you're working on this plugin, it's such a great addition to the already amazing tool that Grafana is.

**Reply by washcroft** [reply]
@andnofence what custom plugin are you using to do this, or is it something you created yourself? I'm looking to do this very thing - plot polygons stored in a database on the map.

**Comment by jpaudoin** [comment]
Hi everyone. This new Geomap panel is really great.
However, as geographic database can contain a huge number of records, this quickly overloads the browser.
It could be great to filter the query based on the area displayed on the map.
As we have time filter variables ($__xx) from the time picker, could we have geographic variables for this visualisation used to filter the records based on the zoom level and on the displayed area?
Drawback: this would implies that the query is updated when zooming in/out.
Thanks!

**Comment by kanor10** [comment]
I am using Grafana 10.0.1, how do I enable the alpha geomap layers? The keywords for the .ini file are nowhere to be found. I would greatly appreciate some direction as I am trying to use the route layer

**Reply by Daddaniele** [reply]
Hi,
I asked the same question on 'other channels' and received the following:

> The Route layer is still an alpha feature, so you have to enable it. To do so, add the `alphaPanels` feature toggle in your `grafana.ini` file, like so:
> 
> [feature_toggles]
> //enable features, separated by spaces
> enable = alphaPanels topnav topNavCommandPalette newPanelChromeUI dataConnectionsConsole

I've tried but at the moment I still can't see the option. ( maybe there's something else or maybe I am just unlucky, dunno ..  so give it a try )

**[ UPDATE ]**
same discussion, new solution ( this one works for me )
On grafana.ini 
put this new lines

[panels]
enable_alpha = true

restart grafana server and here you are.
( if u are reading this, thank you, Isabel ).

**Comment by teepox** [comment]
Support for displaying raw values on the map would be nice. I'd like to visualize environment sensor data from several locations (think "weather forecast map"), but haven't found a good way to do that.

**Comment by Daddaniele** [comment]
Hi,
i'd like to have a "different route layer":
the existing one takes a list of points ( lat,lng ) and concatenate all the points to draw a route.
The first point is the source, the last point is the destination of the single route.

what i have is a large list of sources ( let's say 1k lat,lng of source ) and a few destinations ( let's say 10 lat2,lng2 ) and i would like to draw a line from each source to his own destination ( a straight line, a slightly curved line .. doesn't really matter. ).

**Comment by tobiasdirksen** [comment]
A very important suggestion in my opinion, is a possibility to specify a click action such that clicking on an element changes the value of a variable.

**Comment by mas-kon** [comment]
Hi. I have DB ClickHouse. In my table I have fields
posLatitude as Float64 and posLongitude as Float64.
But geomap was write "No numbered fields found". 
What's wrong?

**Comment by xaxeu** [comment]
+1 for aditional layer with weather information.
The use case is to aggregate in the same map:
- track route - already supported
- photos taken in that path - already supported
- weather conditions in that path, based on location and date. 
  - Not sure which free weather services can provide this info online, but would be great to have conditions like wind speed, wind direction, tides/currents (great for those who do aquatic sports), temp, hum, etc

**Comment by FZubietaC** [comment]
Hi! I'm new at Geomap and i'd like to know if it can allow Shapefiles or geometry tables.

**Comment by dunklesToast** [comment]
Just started using the GeoMap feature and it's awesome but theres one feature I'd like:
If I have a GeoJSON Layer it'd be great to have an option to refetch the GeoJSON file when the dashboard refreshes. This way it'd be possible to have some remote service update the GeoJSON file and always show the most up to date version in Grafana.

**Comment by florentaden** [comment]
Hi! Thank you for the plugging, it's very useful. I am wondering if there is a way to get the boundaries of the map under variables that can be used to filter out some part of a query. So, when I zoom in/out it changes the boundaries variables and update the query result. That would be awesome! Cheers

**Comment by jameshueston** [comment]
@drew08t does the Geomap panel currently support geofences (polygons) with the ability to alert/take action when a point is received inside the fence?  if not currently supported, is it planned?

Are there other well-tested Grafana plugins that would support geofences with alerts that work with Grafana Cloud Advanced or Cloud Enterprise?

**Comment by athornboroughp4** [comment]
Hey @drew08t! Thank you for letting the community to give feedback on the panel. 
First off I want to say I really enjoy the panel and have been able to create some cool things with it :)

A scenario that would be awesome to achieve is the following:

The high level would be to highlight data points in the Geomap panel based on other time series data as a product of time/shared crosshair.  Then you can correlate timeseries data with geomap/location data.
 
Imagine if you have some temperature sensors on a device that also moves around the map in the geomap panel (For example, a ice cream truck). It would be great to correlate where the device was on the map with the temperatures data that was recorded based on that given timestamp.  So then we can visually see in this city at this timestamp the temperatures was x. While that night in the next city over the temperature was y.

Visually how that is represented could be the geolocation data being highlighted (a brighter hue or a different hue compared to the other data ) compared to a time series graph via shared crosshair. So essentially the shared crosshair in the time series panel will highlight different data points on the geomap based on other timeseries data. 

That way we can correlate visually geolocation data to time series data, giving us a better picture of a application based on geolocation.

Let me know if you have any questions or I may have missed a feature already baked in.

Thanks again for the great panel!

**Comment by natellium** [comment]
Hey there! We want to let you all know that we are moving to Github Issues for feature requests. You can read more about the move [here](https://github.com/grafana/grafana/discussions/73424).

I will be closing this discussion, but please don’t hesitate to open a [feature request](https://github.com/grafana/grafana/issues/new?assignees=&labels=&projects=&template=1-feature_requests.md) with any further feedback/suggestions. Sorry for the inconvenience, and thanks for your contributions!

cc @grafana/dataviz-squad
